tests/test_search_dashboard.py

```diff
@@ -0,0 +1,209 @@
+import functools
+import pytest
+from collections import namedtuple
+from pyspark.sql.types import (
+    StructField, ArrayType, StringType, LongType, StructType, DoubleType
+)
+from mozetl.search.dashboard import (
+    search_dashboard_etl, explode_search_counts, add_derived_columns
+)
+
+
+# Some boilerplate to help define example dataframes for testing
+
+# A helper class for declaratively creating dataframe factories
+dataframe_field = namedtuple('dataframe_field', [
+    'name',
+    'default_value',
+    'type',
+    'nullable',
+])
+
+
+def to_field(field_tuple):
+    """Create a dataframe_field from a tuple"""
+    return dataframe_field(*field_tuple)
+
+
+def get_dataframe_factory_config(fields):
+    """Parse a list of dataframe_fields to a schema and set of default values"""
+    schema = StructType([
+        StructField(field.name, field.type, field.nullable)
+        for field in fields
+    ])
+    default_sample = {field.name: field.default_value for field in fields}
+
+    return schema, default_sample
+
+
+@pytest.fixture
+def define_dataframe_factory(dataframe_factory):
+    def partial(fields):
+        """Create a dataframe_factory from a set of field configs"""
+        schema, default_sample = get_dataframe_factory_config(fields)
+        return functools.partial(
+            dataframe_factory.create_dataframe,
+            base=default_sample,
+            schema=schema
+        )
+
+    return partial
+
+
+# Boilerplate for generating example main_summary tables
+def generate_search_count(engine='google', source='urlbar', count=4):
+    return {
+        'engine': engine,
+        'source': source,
+        'count':  count,
+    }
+
+
+@pytest.fixture()
+def generate_main_summary_data(define_dataframe_factory):
+    return define_dataframe_factory(map(to_field, [
```

Comment:
I am torn. I had it inside `define_dataframe_factory` but decide to remove it. The `dataframe_field` class serves two purposes: 1) allowing easy access to each field and 2) documenting what each field means.

I was concerned the nameless tuples would be difficult to understand without keeping `to_field` close to the tuple definition. If you don't think this helps readability, I'm happy to move this back into `define_dataframe_factory`